File: app/src/main/java/com/o4x/musical/ui/activities/tageditor/ArtistTagEditorActivity.java
package com.o4x.musical.ui.activities.tageditor;

import android.content.Intent;

import androidx.annotation.NonNull;

import com.o4x.musical.R;
import com.o4x.musical.loader.ArtistLoader;
import com.o4x.musical.model.Song;
import com.o4x.musical.network.Models.DeezerArtistModel;
import com.o4x.musical.ui.activities.tageditor.onlinesearch.ArtistSearchActivity;

import java.util.ArrayList;
import java.util.List;

public class ArtistTagEditorActivity extends AbsTagEditorActivity<DeezerArtistModel.Data> {

    @Override
    protected int getContentViewLayout() {
        return R.layout.activity_artist_tag_editor;
    }

    @NonNull
    @Override
    protected List<String> getSongPaths() {
        List<Song> songs = ArtistLoader.getArtist(this, getId()).getSongs();
        List<String> paths = new ArrayList<>(songs.size());
        for (Song song : songs) {
            paths.add(song.data);
        }
        return paths;
    }

    @Override
    protected void fillViewsWithResult(DeezerArtistModel.Data result) {
        loadImageFromUrl(result.pictureXl, result.name);
        if (artistName != null)
            artistName.setText(result.name);
    }

    @Override
    protected void searchImageOnWeb() {
        searchWebFor(artistName.getText().toString());
    }

    @Override
    protected void searchOnline() {
        Intent intent = new Intent(this, ArtistSearchActivity.class);
        intent.putExtra(ArtistSearchActivity.EXTRA_SONG_NAME, tagUtil.getArtistName());
        this.startActivityForResult(intent, ArtistSearchActivity.REQUEST_CODE);
    }

    @Override
    protected void getImageFromLastFM() {
        
    }

}

File: app/src/main/java/com/o4x/musical/ui/activities/tageditor/ArtistTagEditorActivity.kt
package com.o4x.musical.ui.activities.tageditor

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.widget.FrameLayout
import com.o4x.musical.extensions.hide
import com.o4x.musical.extensions.show
import com.o4x.musical.model.Artist
import com.o4x.musical.network.Models.DeezerArtistModel
import com.o4x.musical.ui.activities.tageditor.onlinesearch.ArtistSearchActivity
import java.util.*

class ArtistTagEditorActivity : AbsTagEditorActivity<DeezerArtistModel.Data>() {

    override fun albumImageView() = binding.frontImage
    override fun artistImageView() = binding.backImage

    override fun showViews() {
        binding.apply {
            artist.show()
            frontImage.hide()
        }
    }

    override fun createArtist(): Artist = repository.artistById(id)

    override fun createPaths(): List<String> {
        val songs = artist.songs
        val paths: MutableList<String> = ArrayList(songs.size)
        for (song in songs) {
            paths.add(song.data)
        }
        return paths
    }

    override fun fillViewsWithResult(result: DeezerArtistModel.Data) {
        loadImageFromUrl(result.pictureXl, null)
        binding.artist.editText?.setText(result.name)
    }

    override fun searchImageOnWeb() {
        searchWebFor(binding.artist.editText?.text.toString())
    }

    override fun searchOnline() {
        val intent = Intent(this, ArtistSearchActivity::class.java)
        intent.putExtra(ArtistSearchActivity.EXTRA_SONG_NAME, tagUtil!!.artistName)
        this.startActivityForResult(intent, ArtistSearchActivity.REQUEST_CODE)
    }
}